tutorials/best_practices/godot_notifications.rst

```diff
@@ -164,68 +164,64 @@ _init vs. initialization vs. export
 -----------------------------------
 
 If the script initializes its own node subtree, without a scene,
-that code should execute here. Other property or SceneTree-independent
-initializations should also run here. This triggers before ``_ready()`` or
-``_enter_tree()``, but after a script creates and initializes its properties.
+that code should execute in ``_init()``. Other property or SceneTree-independent
+initializations should also run here.
 
-Scripts have three types of property assignments that can occur during
-instantiation:
+``_init()`` triggers before ``_enter_tree()`` or ``_ready()``, but after a script
+creates and initializes its properties.  When instantiating a scene, property 
+values will set up according to the following sequence:
+
+1. **Initial value assignment:** the property is assigned its initialization value,
+   or ``null`` if one is not specified.  If a setter exists, it is not used.
+
+2. **``_init()`` assignment:** the property's value is replaced by any assignments
+   made in ``_init()``, triggering the setter.
+
+3. **Exported value assignment:** an exported property's value is again replaced by
+   any value set in the Inspector, triggering the setter.
 
 .. tabs::
   .. code-tab:: gdscript GDScript
 
-    # "one" is an "initialized value". These DO NOT trigger the setter.
-    # If someone set the value as "two" from the Inspector, this would be an
-    # "exported value". These DO trigger the setter.
+    # test is initialized to "one", without triggering the setter.
     @export var test: String = "one":
         set(value):
-            test = value
-            print("Setting: ", test)
+            test = value + "!"
 
     func _init():
-        # "three" is an "init assignment value".
-        # Trigger the setter
-        test = "three"
+        # Triggers the setter, changing test's value from "one" to "two!".
+        test = "two"
+
+    # If someone sets test to "three" from the Inspector, it would trigger
+    # the setter, changing test's value from "two!" to "three!".
 
   .. code-tab:: csharp
 
     using Godot;
 
     public partial class MyNode : Node
     {
+        // _test is initialized to "one".
```

Comment:
Fair. I figured that since Godot is aimed at beginner programmers, they may be using C# for the first time (and probably should have chosen GDScript for their first project.)  In that case they wouldn't necessarily understand how initializers vs constructors worked.

I wanted to point out that it is in fact initialized even when it's set in the constructor, because if you say ``MyObject _test = new MyObject();`` then the side effects of object creation still happen.

Reply:
The C# documentation expects users to be somewhat familiar with C# already, and if they aren't there are links to the Microsoft documentation. It would be a daunting task for the Godot documentation to also document every C# feature, and we let external sources handle it.

Reply:
Ah, I hadn't known that, forgive the oversight.  I'll update appropriately.

Reply:
I updated from command line rather than direct edits on Github this time so I didn't get a Verified badge next to my commit, but I hope it still works.  I believe I've addressed all comments now, please let me know if you'd like anything else changed!

Reply:
Once you've addressed all feedback (if there's any more after this) please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

Reply:
Thanks for the pointer and sorry for the bad workflow.  I believe I've correctly rebased the PR off the latest upstream master, and I'm not aware of any more feedback.